implementations/rust/ockam/ockam_node/src/storage/database/migrations/migration_support/migrator.rs

```diff
@@ -266,19 +187,262 @@ impl Migrator {
+    fn partial_cmp(&self, other: &Self) -> Option<Ordering> {
+        Some(self.cmp(other))
+    }
+}
+
+#[cfg(test)]
+mod tests {
+    use super::*;
+    use ockam_core::async_trait;
+    use sqlx::migrate::MigrationType;
+
+    #[test]
+    fn version_inside_interval() {
+        let interval = VersionInterval::from_to(1, 3);
+        assert!(!interval.version_is_inside(0));
+        assert!(interval.version_is_inside(1));
+        assert!(interval.version_is_inside(2));
+        assert!(!interval.version_is_inside(3));
+        assert!(!interval.version_is_inside(4));
+
+        let interval = VersionInterval::from_to_inclusive(1, 3);
+        assert!(!interval.version_is_inside(0));
+        assert!(interval.version_is_inside(1));
+        assert!(interval.version_is_inside(2));
+        assert!(interval.version_is_inside(3));
+        assert!(!interval.version_is_inside(4));
+    }
+
+    #[test]
+    fn ordering_of_migrations() {
+        let sql_1 = SqlxMigration::new(1, "sql_1".into(), MigrationType::Simple, "1".into());
+        let sql_2 = SqlxMigration::new(2, "sql_2".into(), MigrationType::Simple, "2".into());
+        let rust_1: Box<dyn RustMigration> = Box::new(DummyRustMigration1);
+        let rust_2: Box<dyn RustMigration> = Box::new(DummyRustMigration2);
+        let rust_3: Box<dyn RustMigration> = Box::new(DummyRustMigration3);
+
+        let mut migrations = vec![
+            NextMigration::Sql(&sql_2),
+            NextMigration::Sql(&sql_1),
+            NextMigration::Rust(rust_1.as_ref()),
+            NextMigration::Rust(rust_3.as_ref()),
+            NextMigration::Rust(rust_2.as_ref()),
+        ];
+        migrations.sort();
+
+        for m in migrations.iter() {
+            match m {
+                NextMigration::Sql(_) => {
+                    assert!(m.is_sql());
+                }
+                NextMigration::Rust(_) => {
+                    assert!(!m.is_sql());
+                }
+            }
+        }
+
+        assert_eq!(
+            migrations,
+            vec![
+                NextMigration::Sql(&sql_1),
+                NextMigration::Rust(rust_1.as_ref()),
+                NextMigration::Sql(&sql_2),
+                NextMigration::Rust(rust_2.as_ref()),
+                NextMigration::Rust(rust_3.as_ref())
+            ]
+        );
+    }
+
+    macro_rules! impl_rust_migration {
+        ($name:ident, $version:expr) => {
+            #[derive(Debug)]
+            struct $name;
+
+            #[async_trait]
+            impl RustMigration for $name {
+                fn name(&self) -> &str {
+                    stringify!($name)
+                }
+
+                fn version(&self) -> Version {
+                    $version
+                }
+
+                async fn migrate(&self, _connection: &mut SqliteConnection) -> Result<bool> {
+                    Ok(true)
+                }
+            }
+        };
+    }
+
+    impl_rust_migration!(DummyRustMigration1, 1);
```

Comment:
Wouldn't it be easier to define `DummyRustMigration` that takes one number in the constructor? But up to you

Reply:
Totally, I did this thinking that I might need a unique name as well and didn't review it after writing the tests. Thanks!